frontend/www/js/omegaup/course/edit.ts

```diff
@@ -172,16 +173,47 @@ OmegaUp.on('ready', () => {
               })
               .catch(ui.apiError);
           },
+          'get-versions': (
+            problemAlias: string,
+            source: course_ProblemList,
+          ) => {
+            api.Problem.versions({ problem_alias: problemAlias })
+              .then(function (result) {
+                source.versionLog = result.log;
+                let currentProblem = null;
+                for (const problem of source.problems) {
+                  if (problem.alias === problemAlias) {
+                    currentProblem = problem;
+                    break;
+                  }
+                }
+                let publishedCommitHash = result.published;
+                if (currentProblem != null) {
+                  publishedCommitHash = currentProblem.commit;
+                }
+                for (const revision of result.log) {
+                  if (publishedCommitHash === revision.commit) {
+                    source.selectedRevision = source.publishedRevision = revision;
+                    break;
+                  }
+                }
+              })
+              .catch(ui.apiError);
+          },
           'add-problem': (
             assignment: types.CourseAssignment,
             problem: types.AddedProblem,
           ) => {
-            api.Course.addProblem({
+            const problemParams = {
               course_alias: courseAlias,
               assignment_alias: assignment.alias,
               problem_alias: problem.alias,
               points: problem.points,
-            })
+            };
+            if (problem.commit) {
+              Object.assign(problemParams, { commit: problem.commit });
```

Comment:
```suggestion
              problemParams.commit = problem.commit;
```

esto funciona? `Object.assign` funciona para asignar muchas cosas a la vez, pero si es sólo una, posiblemente sea mejor preferir la versión más simple.

Reply:
Si funciona. Se cambia